pandas/core/apply.py

```diff
@@ -204,16 +204,19 @@ def apply_empty_result(self):
         from pandas import Series
 
         if not reduce:
-
-            EMPTY_SERIES = Series([])
             try:
-                r = self.f(EMPTY_SERIES, *self.args, **self.kwds)
+                r = self.f(Series([]))
                 reduce = not isinstance(r, Series)
             except Exception:
                 pass
 
         if reduce:
-            return self.obj._constructor_sliced(np.nan, index=self.agg_axis)
+            if len(self.agg_axis):
+                r = self.f(Series([]))
```

Comment:
we *already* are trying to reduce above (line 208), why are you calling the function again? does this hit the Except?

Reply:
I did that in case `reduce` was already `True` at line 206, so that the `try` block wouldn't have been executed

Reply:
i am still puzzled why you can not pass args/kwargs

Reply:
I think what's happening is the function `self.f` is getting curried around here: https://github.com/pandas-dev/pandas/blob/cb68153f4b6fca19440ec6b79a0d1128c002ec11/pandas/core/apply.py#L109 , so when we pass the arguments _again_ we get an error if we try to call `df.nunique()`:
```
>           r = self.f(Series([]), *self.args, **self.kwds)
E           TypeError: f() got an unexpected keyword argument 'dropna'
```
because at that point `f` only takes a single argument.  I could imagine there could end up being a problem if this currying doesn't happen, so there's probably a hidden corner case that just isn't covered by the existing tests.

Reply:
Was confused by this to but I think this makes sense. I suppose this was hitting the `except` before due to a `TypeError` for wrong number of arguments?